pulp_smash/tests/rpm/api_v2/test_orphan_remove.py

```diff
@@ -0,0 +1,79 @@
+# coding=utf-8
+"""Test the removal of rpm orphans.
+
+This module assumes that the tests in
+:mod:`pulp_smash.tests.platform.api_v2.test_repository` and
++`pulp_smash.tests.platform.api_v2.test_sync_publish` hold true.
+"""
+
+try:
+    from urllib.parse import urljoin
+except ImportError:
+    from urlparse import urljoin  # pylint:disable=C0411,E0401
+
+from pulp_smash import api, utils
+from pulp_smash.constants import (
+    ORPHANS_PATH,
+    REPOSITORY_PATH,
+    RPM_FEED_URL,
+)
+
+
+def _gen_repo():
+    """Return a semi-random dict for use in creating an RPM repository."""
+    return {
+        'id': utils.uuid4(),
+        'importer_config': {},
+        'importer_type_id': 'yum_importer',
+        'notes': {'_repo-type': 'rpm-repo'},
+    }
+
+
+def _count_orphans(client):
+    """Count the number of orphans."""
+    orphan_dict = client.get(ORPHANS_PATH)
+    total_orphans = 0
+    for _, data in orphan_dict.items():
+        total_orphans += data.get('count') or 0
+    return total_orphans
+
+
+class OrphanRemoveAllTestCase(utils.BaseAPITestCase):
+    """Establish that orphan removal does a full cleanup."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Create an RPM repo, sync it, delete the repo, remove orphans."""
+        super(OrphanRemoveAllTestCase, cls).setUpClass()
+        client = api.Client(cls.cfg, api.json_handler)
+        body = _gen_repo()
+        body['importer_config']['feed'] = RPM_FEED_URL
+        repo = client.post(REPOSITORY_PATH, body)
+        client.response_handler = api.echo_handler
+        sync_path = urljoin(repo['_href'], 'actions/sync/')
+        cls.report = client.post(sync_path, {'override_config': {}})
```

Comment:
I see we're making `report` a class attribute here, but we're not actually using this attribute in any other methods.